### ISSUE TITLE
Wrapping & indent broken when used together for non-monospace glyphs (proportional fonts or non-latin chars)

### ISSUE THREAD

**Issue by byteit101** [microsoft/vscode#132220]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.59.0
- OS Version: Debian 10
- Monaco Version: 0.27.0

Steps to Reproduce VS Code:

1. Set Wrapping Indent to Indent and Wrapping Strategy to Simple (default).
2. Make a line with wider/narrower characters, note it doesn't wrap at the right-border (after/before, respectively) but does indent.
3. Change Wrapping Strategy to Advanced.
4. The line now wraps at the right-border, but no longer indents.
5. Expected: The line wraps at the right-border and indents one tab after doing so

Monaco Playground script:

```js
var editor = monaco.editor.create(document.getElementById("container"), {
	value: "// W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W W  \n//i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i i ",
    fontFamily: 'Arial' // not required, but shows how the "simple" strategy fails for non-monospaced things
    wordWrap: 'on',
    wrappingIndent: "indent",

    // wrappingStrategy: 'simple', // incorrect wrapping, correct indents
    wrappingStrategy: 'advanced', // incorrect indents, correct wrapping
});
```

Separately, the indent and the wrapping work. Together, they fail.

Given that both halves of this (indent, wrap-breakpoint) work when the other is disabled, it looks like a bug to me.

**Comment by gjsjohnmurray** [comment]
Does `"editor.wrappingStrategy": "advanced"` help you?

**Comment by alexdima** [comment]
`wrappingIndent: "indent"` is currently not compatible with `wrappingStrategy: 'advanced'`. 

The reason is the following:

`wrappingStrategy: 'advanced'` means that we delegate the wrap points computations to the browser. This works by us creating a dom node for each line of the editor, setting the contents of each dom node to the text on each line, and then setting a specific width. We then proceed to use browser APIs to detect the wrapping points. This is super slow (causes a forced browser layout).

To add support for `wrappingIndent: "indent"`, we would need to do the above twice: the first time with the dom nodes set to the width of the editor, then detect the wrapping points of each first line, and then repeat this process again with all the dangling lines, but this time with a dom node width smaller than the editor width (i.e. editor width - indent width). This would be double super slow (it would cause a forced browser layout again). 

We have therefore opted to not support `wrappingIndent: "indent"` with `wrappingStrategy: 'advanced'`.

**Comment by byteit101** [comment]
@alexdima Wouldn't doing something like `<div style="text-indent: -4ch; padding-left: 4ch;">` on the DOM node allow it to be done in one pass?

**Comment by alexdima** [comment]
@byteit101 Yes, indeed, that seems to do the trick! (I tried at https://jsfiddle.net/58ys1mjf/ ). Would you be interested in contributing this as a PR. The relevant `<div>` building is [here](https://github.com/microsoft/vscode/blob/f6704bfe7dcaf62cca5e49cade279d9bb85a3799/src/vs/editor/browser/view/domLineBreaksComputer.ts#L193) and that method could get passed in the `wrappingIndent`, and depending on that use your technique.

**Comment by byteit101** [comment]
Well, that was a trivial fix. Only change from my suggestion is to use the size of spaces instead of ch. #134171
